README.md

```diff
@@ -26,9 +26,86 @@ NOTE: at present, only some components have transitioned to this new model, the
   * If the admin credentials cannot create additional credentials, but do themselves fulfill the requirements of the credentials request, they will be used. (with logged warnings and a condition on the credentials request)
   * If the admin credentials fulfill neither of the above requirements, the controller will fail to generate the credentials, report failure back to the Cluster Version Operator, and thus block upgrading. The installer will also perform this check early to inform the user their cluster will not function.
 
-## AWS
-
-Currently AWS is the only supported cloud provider. If running in-cluster, your credentials are expected to exist in kube-system/aws-creds with the keys aws_access_key_id and aws_secret_access_key.
+## Cloud Credentials
+If running in-cluster, your credentials are expected to exist in kube-system namespace.
+Credentials stored in a secret that is created per deployment infrastructure.
+Here are the available options:
+  1. AWS:
+  ```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: kube-system
+    name: aws-creds
+  data:
+    aws_access_key_id: Base64encodeAccessKeyID
+    aws_secret_access_key: Base64encodeSecretAccessKey
+  ```
+  2. Azure:
+  ```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: kube-system
+    name: azure-credentials
+  data:
+    azure_subscription_id: Base64encodeSubscriptionID
+    azure_client_id: Base64encodeClientID
+    azure_client_secret: Base64encodeClientSecret
+    azure_tenant_id: Base64encodeTenantID
+    azure_resource_prefix: Base64encodeResourcePrefix
+    azure_resourcegroup: Base64encodeResourceGroup
+    azure_region: Base64encodeRegion
+  ```
+  3. GCP:
+  ```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: kube-system
+    name: gcp-credentials
+  data:
+    service_account.json: Base64encodeServiceAccount
+  ```
+  4. OpenStack:
+  ```yaml
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: kube-system
+    name: openstack-credentials
+  data:
+    clouds.yaml: Base64encodeCloudCreds
+    clouds.conf: Base64encodeCloudCredsINI
+  ```
+  5. VSphere:
```

Comment:
so close to alphabetical. can we move VSphere to the bottom of the list?

Reply:
@joelddiaz 
Done.